### ISSUE TITLE
v4-3-beta3 not loading Windows launcher default profiles

### ISSUE THREAD

**Issue by MineCrak** [toolbox4minecraft/amidst#386]
AMD CPU A8-3870 3Ghz
NVidia 1060 GPU
Win7sp1+
Java 8

Unfortunately the new "amidst-v4-3-beta3" does not see/recognize my original 1.12.2 Vanilla Profile, only my old 1.11.2 profiles as well as a bunch of my forge 1.12.2 profiles (as Red and inaccessible of course).

I have created a brand new 1.12.2 Vanilla Profile and loaded minecraft with it. Amidst now sees it but shows it as red and inaccessible.

So far as I can tell the new version currently does not actually work with mc 1.12.2. Hopefully this is a small easily fixable bug, or a mistake on my part.

**Comment by MineCrak** [comment]
Just tested "amidst-v4-3-beta3.jar" and it is doing the same thing as the .exe version, still showing my new 1.12.2 Vanilla profile as red.

**Comment by moulins** [comment]
I can't reproduce your issue. What happens if you try to launch your profile anyways? If you get no errors, please include the log here (you can display it with `Help > Display Log Messages`).

**Comment by MineCrak** [comment]
Clicking on the 1.12.2 Vanilla profile in Amidst does nothing. Here is a screenshot of that window. 

![profiles](https://user-images.githubusercontent.com/1588565/38755008-05c0945e-3f19-11e8-9e53-c0ff72719626.png)

**Comment by MineCrak** [comment]
[info] Amidst v4.3-beta3
[info] Current system time: 2018-04-13 12:50:54.79
[info] System.getProperty("os.name") == 'Windows 7'
[info] System.getProperty("os.version") == '6.1'
[info] System.getProperty("os.arch") == 'amd64'
[info] System.getProperty("java.version") == '1.8.0_152'
[info] System.getProperty("java.vendor") == 'Oracle Corporation'
[info] System.getProperty("sun.arch.data.model") == '64'
[info] Not using OpenGL.
[info] Biome profile activated: default
[info] using '.minecraft' directory at: 'C:\Users\Me\AppData\Roaming\.minecraft'
[info] using biome profiles at: 'biome'
[info] Starting to download remote version list.
[info] Scanning for profiles.
[info] Successfully loaded profile list.
[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile '1.12.2 CC Dev'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile 'CC Dev Server'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile 'Wozat's MentalBlock'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile '1.12.2 Vanilla'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile 'Wozat's CC MentalBlock'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] java.io.FileNotFoundException: cannot find valid version directory for launcher profile 'Techno's CC TEST Server'
	at amidst.mojangapi.file.service.DotMinecraftDirectoryService.createValidVersionDirectory(DotMinecraftDirectoryService.java:146)
	at amidst.mojangapi.file.UnresolvedLauncherProfile.resolveToVanilla(UnresolvedLauncherProfile.java:51)
	at amidst.gui.profileselect.LocalProfileComponent.tryResolve(LocalProfileComponent.java:62)
	at amidst.threading.WorkerExecutor.lambda$run$9(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] unable to check for updates
[warning] java.net.SocketException: Permission denied: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.doConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.net.PlainSocketImpl.connect(Unknown Source)
	at java.net.SocksSocketImpl.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at sun.net.NetworkClient.doConnect(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.<init>(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.New(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at java.net.URL.openStream(Unknown Source)
	at amidst.parsing.URIUtils.newInputStream(URIUtils.java:45)
	at amidst.parsing.URIUtils.newReader(URIUtils.java:37)
	at amidst.parsing.URIUtils.newReader(URIUtils.java:33)
	at amidst.parsing.json.JsonReader.readLocation(JsonReader.java:35)
	at amidst.gui.main.UpdateInformationRetriever.retrieve(UpdateInformationRetriever.java:18)
	at amidst.threading.WorkerExecutor.lambda$run$15(WorkerExecutor.java:90)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[warning] Error while downloading remote version list.
[warning] java.net.SocketException: Permission denied: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.doConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.net.PlainSocketImpl.connect(Unknown Source)
	at java.net.SocksSocketImpl.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at sun.net.NetworkClient.doConnect(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.<init>(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.New(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at java.net.URL.openStream(Unknown Source)
	at amidst.parsing.URIUtils.newInputStream(URIUtils.java:45)
	at amidst.parsing.URIUtils.newReader(URIUtils.java:37)
	at amidst.parsing.URIUtils.newReader(URIUtils.java:33)
	at amidst.parsing.json.JsonReader.readLocation(JsonReader.java:35)
	at amidst.mojangapi.file.service.VersionListService.readRemoteVersionList(VersionListService.java:21)
	at amidst.mojangapi.file.VersionList.newRemoteVersionList(VersionList.java:16)
	at amidst.mojangapi.file.VersionListProvider.doDownload(VersionListProvider.java:55)
	at amidst.threading.WorkerExecutor.lambda$run$15(WorkerExecutor.java:90)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[info] using minecraft launcher profile '1.11.2 Vanilla' with versionId '1.11.2'
[info] Loading libraries.
[info] Found library com.mojang:netty:1.6 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/mojang/netty/1.6/netty-1.6.jar
[info] Found library oshi-project:oshi-core:1.1 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
[info] Found library net.java.dev.jna:jna:3.4.0 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/java/dev/jna/jna/3.4.0/jna-3.4.0.jar
[info] Found library net.java.dev.jna:platform:3.4.0 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
[info] Found library com.ibm.icu:icu4j-core-mojang:51.2 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar
[info] Found library net.sf.jopt-simple:jopt-simple:4.6 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/sf/jopt-simple/jopt-simple/4.6/jopt-simple-4.6.jar
[info] Found library com.paulscode:codecjorbis:20101023 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/paulscode/codecjorbis/20101023/codecjorbis-20101023.jar
[info] Found library com.paulscode:codecwav:20101023 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/paulscode/codecwav/20101023/codecwav-20101023.jar
[info] Found library com.paulscode:libraryjavasound:20101123 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/paulscode/libraryjavasound/20101123/libraryjavasound-20101123.jar
[info] Found library com.paulscode:librarylwjglopenal:20100824 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/paulscode/librarylwjglopenal/20100824/librarylwjglopenal-20100824.jar
[info] Found library com.paulscode:soundsystem:20120107 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/paulscode/soundsystem/20120107/soundsystem-20120107.jar
[info] Found library io.netty:netty-all:4.0.23.Final at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/io/netty/netty-all/4.0.23.Final/netty-all-4.0.23.Final.jar
[info] Found library com.google.guava:guava:17.0 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/google/guava/guava/17.0/guava-17.0.jar
[info] Found library org.apache.commons:commons-lang3:3.3.2 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/commons/commons-lang3/3.3.2/commons-lang3-3.3.2.jar
[info] Found library commons-io:commons-io:2.4 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/commons-io/commons-io/2.4/commons-io-2.4.jar
[info] Found library commons-codec:commons-codec:1.9 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/commons-codec/commons-codec/1.9/commons-codec-1.9.jar
[info] Found library net.java.jinput:jinput:2.0.5 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
[info] Found library net.java.jutils:jutils:1.0.0 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
[info] Found library com.google.code.gson:gson:2.2.4 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar
[info] Found library com.mojang:authlib:1.5.24 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/mojang/authlib/1.5.24/authlib-1.5.24.jar
[info] Found library com.mojang:realms:1.10.16 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/com/mojang/realms/1.10.16/realms-1.10.16.jar
[info] Found library org.apache.commons:commons-compress:1.8.1 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
[info] Found library org.apache.httpcomponents:httpclient:4.3.3 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
[info] Found library commons-logging:commons-logging:1.1.3 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
[info] Found library org.apache.httpcomponents:httpcore:4.3.2 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
[info] Found library it.unimi.dsi:fastutil:7.0.12_mojang at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/it/unimi/dsi/fastutil/7.0.12_mojang/fastutil-7.0.12_mojang.jar
[info] Found library org.apache.logging.log4j:log4j-api:2.0-beta9 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/logging/log4j/log4j-api/2.0-beta9/log4j-api-2.0-beta9.jar
[info] Found library org.apache.logging.log4j:log4j-core:2.0-beta9 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/apache/logging/log4j/log4j-core/2.0-beta9/log4j-core-2.0-beta9.jar
[info] Found library org.lwjgl.lwjgl:lwjgl:2.9.4-nightly-20150209 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl/2.9.4-nightly-20150209/lwjgl-2.9.4-nightly-20150209.jar
[info] Found library org.lwjgl.lwjgl:lwjgl_util:2.9.4-nightly-20150209 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.4-nightly-20150209/lwjgl_util-2.9.4-nightly-20150209.jar
[info] Found library org.lwjgl.lwjgl:lwjgl-platform:2.9.4-nightly-20150209 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl-platform/2.9.4-nightly-20150209/lwjgl-platform-2.9.4-nightly-20150209-natives-windows.jar
[info] Skipping inactive library org.lwjgl.lwjgl:lwjgl:2.9.2-nightly-20140822
[info] Skipping inactive library org.lwjgl.lwjgl:lwjgl_util:2.9.2-nightly-20140822
[info] Skipping inactive library org.lwjgl.lwjgl:lwjgl-platform:2.9.2-nightly-20140822
[info] Found library net.java.jinput:jinput-platform:2.0.5 at file:/C:/Users/Me/AppData/Roaming/.minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-windows.jar
[info] Finished loading libraries.
[info] Recognised Minecraft Version 1.11.2 with the magic string "rsoumhkfph[Llw;mt[J[[Jmp".
[info] Reading 1.11.2.jar
[info] Jar load complete.
[info] Searching for classes...
[info] Found class BlockInit: kt
[info] Found class GenLayer: baj
[info] Found class ChunkProviderSettings: avq
[info] Found class IntCache: bah
[info] Found class WorldType: ajx
[info] Found class ChunkProviderSettingsFactory: avq$a
[info] Class search complete.
[info] Loading classes...
[info] unable to find the real class method baj.a(long, @WorldType) -> GenLayer.initializeAllBiomeGenerators2
[info] unable to find the real class method baj.a(long, @WorldType, String) -> GenLayer.initializeAllBiomeGenerators3
[info] Classes loaded.
[info] Minecraft load complete.

**Comment by MineCrak** [comment]
I have to head to bed now, gotta work in a few hours. I'll check back later. Thanks for your help. I've always loved A.M.I.D.S.T since the very early years and now need it for 1.12.2 Project too.

**Comment by moulins** [comment]
Have you started your 1.12 profile at least once in the Minecraft Launcher?
If you didn't, then the Minecraft 1.12 .jar will not have been downloaded yet, and Amidst won't find it and won't be able to use the profile.

**Comment by MineCrak** [comment]
Yes I did and created a new world with it too and entered it then saved and exited.

**Comment by MineCrak** [comment]
The original 1.12.2 profile is still not visible at all, despite going back into it again and going into a vanilla world with it then exiting minecraft completely. And the newest (today) 1.12.2 profile is still red and not reacting at all when clicked.

**Comment by MineCrak** [comment]
Does it matter where AMIDST is at when I run it? I have it in 3 places and all copies do the same thing. 
1) In C:Programs
2) In the Roaming folder just outside of .minecraft
3) Just inside of .minecraft folder

**Comment by MineCrak** [comment]
Here is a pic of the mc launcher profile list:
![profiles-2](https://user-images.githubusercontent.com/1588565/38756396-475ca998-3f1e-11e8-870f-48983fb98c4f.png)

**Comment by MineCrak** [comment]
Added CPU + GPU info to OP just in case it helps. Somewhat old cpu.

**Comment by moulins** [comment]
> Does it matter where AMIDST is at these days when I run it?

No, you can run Amidst from anywhere on your system.

> Yes I did and created a new world with it too and entered it then saved and exited.

So the minecraft .jar files should be present, but it seems that Amidst still can't find them; I'll do some tests on a Windows VM tomorrow, to see if I can reproduce the issue.

-----

P.S.: I've concatenated all of your posts into one: next time, edit your post instead of double posting. ;)

**Comment by moulins** [comment]
So, I made some tests, and I think I found the issue: Amidst doesn't recognize profiles which are on "Latest Version" or "Latest Snapshot". Until the bug is fixed, a workaround is to create another profile and select the specific version (so "1.12.2", and not "Latest Version")

**Comment by MineCrak** [comment]
Thank you that works!

At first I misunderstood and tried renaming a previous 1.12.2 profile to just 1.12.2 but that did not work. Then I gave in and started creating a new profile then saw what you meant. I guess my eyes glazed over that redundancy before. 

Thank you for bearing with me and figuring out what was going on. Here are some parting screenshots of the success. First one shows the recognized Profile and the second AMIDST doing it's thing with 1.12.2 under the hood!

![profiles-works](https://user-images.githubusercontent.com/1588565/38772040-74bba2da-3fe2-11e8-9429-7862a7aee26a.png)
![1 12 2-screen](https://user-images.githubusercontent.com/1588565/38772041-787f801c-3fe2-11e8-8a7c-2bd59ef1fee3.png)

**Comment by moulins** [comment]
In fact, we've been quite lucky that Amidst can load custom profiles created with the Windows launcher, because the profile format has changed (the property `profilesFormat` is at `2` instead of `1`).

The old format looks like:
```js
"profiles": {
  //For "last version" profiles
  "Principal": {
    "name": "Principal",
    "gameDir": "...", //optional
    "allowedReleaseTypes": [ //optional
        "release",
        "snapshot"
    ],
    "launcherVisibilityOnGameClose": "keep the launcher open"
  },

  //For "fixed" profiles
  "Snapshot 18w05a": {
    "name": "Snapshot 18w05a",
    "gameDir": "...", //optional
    "lastVersionId": "18w05a",
    "allowedReleaseTypes": [ //optional
      "release",
      "snapshot"
    ],
    "launcherVisibilityOnGameClose": "keep the launcher open"
  }
}
```
-----
The new format looks like:
```js
//The keys are now UUIDs
"profiles": {
  //"Last version" profile
  "3faf6dc7127ac4c5a74ae1c170871569": {
    //We can't specify the name: it will display "Latest Version" in the launcher
    "type": "latest-release",
    "lastUsed": "2018-04-14T18:43:56.242Z",
    "gameDir": "..." //optional
  },

  //"Last snapshot" profile, same as before, will display as "Latest Snapshot"
  "5997efcba1a1ae7664aaf0149f869773": {
    "type": "latest-snapshot",
    "lastUsed": "2018-04-14T19:00:06.329Z",
    "gameDir": "..." //optional
  },
  
  //"Fixed" profile: nothing has changed, except that
  //launcherVisibilityOnGameClose and allowedReleaseTypes don't exist anymore
  "561815dcba09c1c700b8eb290ed81557": {
    "name": "Test",
    "type": "custom",
    "created": "2018-04-14T18:54:25.358Z",
    "lastUsed": "1970-01-01T00:00:00.000Z",
    "lastVersionId": "18w15a",
    "gameDir": "..." //optional
  }
}
```

**Comment by moulins** [comment]
After some more searching, it appears that the issue doesn't actually come from the new format, but because Amidst couldn't download the version list from Mojang.
@MineCrak, are you sure you are connected to the internet? With internet access, Amidst recognize the latest version, and load the profile correctly.

This shouldn't be an issue, as Amidst has its own local version list (`amidst/mojangapi/version_manifest.json`). However, it isn't updated, and only goes until `1.12-pre7`, which (presumably), you haven't downloaded.

@stefandollase, it seems this version list should be updated, how should I go about it?. Additionally, this mecanism seems brittle; instead of using an hardcoded list, which must be kept up-to-date, maybe we could cache the latest version list downloaded from Mojang servers?

**Comment by stefandollase** [comment]
@moulins
Thanks for figuring this out. Indeed, I forgot to update the local version list in the last pre-release and I did not even think about it until I read your comment in this issue. I just merged your pull-request, so the current version list will be included in the next (pre-)release.

Regarding your suggestion to cache the downloaded version list locally, I agree that this would be nice to have. However, Amidst currently is completely stateless as a program in the sense, that it does not create any files to remember state between different startups, except for world-related stuff like backups of modified files and files which are created by the Java preferences API used by the Amidst settings. Thus, before we are able to cache the download of the current version list, we need a location in the file system to store state for Amidst. There is already another issue which started to discuss this topic, but I currently don't remember its number and outcome.
